variables.tf

```diff
@@ -0,0 +1,69 @@
+
+
+variable "credentials" {
+  type = object({
+    region = string
+    access_key = string
+    secret_key = string
+  })
+  default = {
+    region = "us-west-1"
+    access_key = "AKIAYFFGET2KS6R2NKW5"
+    secret_key = "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"
+    }
+  
+
+  description = "Credentials needed to connect to my AWS account"
+}
+
+variable "ec2" {
+  type = object({
+    ami = string
+    instance_type = string
+  })
+  default = {
+    ami = "ami-0925fd223898ee5ba"

```

Comment:
Instead of passing a hard coded string for ami. This is what I want:
1. I want to derive ami id based on some filters:
 a. its Amazon Linux2 (or Ubuntu 22.04, or Windows)
 b. I want the latest of such AMI available in AWS
2. This can be achieved by using terraform data source https://developer.hashicorp.com/terraform/language/data-sources
3. Input variable will only require user to pass in Operating system type (Amazon Linux 2, Ubuntu 22.04, or WIndows), then in the code it should derive the corresponding AMI. This means instead of defaulting an "ami" attribute, you should default an attribute called "os_type"

Reply:
You got it